src/main/scala/models/CustomersInterpreter.scala

```diff
@@ -0,0 +1,13 @@
+package models
+
+import cats.Id
+import models.Domain.{Customer, CustomerId}
+import repo.Repository.CustomerRepository
+
+class CustomersInterpreter(repository: CustomerRepository) extends Customers[Id] {
```

Comment:
Intepreter is in the wrong layer, u should not interpret business service but data providers so:
change name `CustomersInterpreter` -> `CustomerService` and remove `extends Customers[Id]`